yapapi/engine.py

```diff
@@ -466,22 +458,72 @@ async def _process_invoices(self) -> None:
             ):
                 break
 
-    async def _check_debit_note_rate(self, act_id: ActivityId, agr_id: AgreementId, job: "Job"):
-        agreement = self._get_agreement_by_id(agr_id)
-        num_notes = self._number_of_debit_notes[act_id]
+    async def _enforce_debit_note_intervals(self, job: "Job", debit_note: DebitNote):
```

Comment:
This function has over 60 lines, please split it to few smaller functions.

Reply:
fair point